### PR TITLE
First working version for Simple CRUD using the Graph API

### PR DESCRIPTION
**Pull request by cristhiank**
@DavideD  this is the new request with all the changes we discussed.
The CRUD test works and I'm using the Graph API. 
I also added the code styling and I did separate commits for each logical feature.... I still have some things to adjust but it's my first approach, let me know what you think.

### CODE REVIEW

**Comment by DavideD** [comment]
Thanks, I'll have a look at this soon

**Comment by cristhiank** [comment]
Thanks @DavideD , in the mean time, I wanted to aak you about how the provider should handle the schema creation, OrientDB supports schema-less, schema-full and hybrid approaches. I was thinking that we should let the user to decide if he wants the provider to create non existent "tables" (classes in OrientDB) or if he wants to create the schema before. Looking forward for your thoughts about this.

**Comment by DavideD** [comment]
An Entity need at least an id, so we can assume that there is always going to be some sort of schema. I think, as a default, we should use a schema-full (or at least hybrid) approach.
JPA already provides many annotations and attributes that can be naturally mapped in constraints over OrientDB: `@Index`, `@Column(nullable, unique)` and so on.

At the moment, it is not possible to add unmapped properties using OGM, we have an issue open related to MongoDB (https://hibernate.atlassian.net/browse/OGM-470) and I guess in OrientDB we might use a similar approach if the user wants.

In OGM, we have the `SchemaDefiner` interface that it is used to get the metadata of the entities and associations and convert them to constraints and definitions on the datastore.

I don't think at the moment we have a property to explicitly disable the schema creation, but we have the following property to decide what to do with unique constraint definitions: `org.hibernate.cfg.AvailableSettingsUNIQUE_CONSTRAINT_SCHEMA_UPDATE_STRATEGY`
(hibernate.schema_update.unique_constraint_strategy)
We use it in `Neo4jSchemaDefiner`.

**Comment by DavideD** [comment]
I didn't mention to you before but we tend not to merge branches unless they are rebased to the latest changes before. This way we always have a linear history log.
No need to change the PR this time, I will just squash the 3 commits together.

File: orientdb/src/test/java/org/hibernate/datastore/ogm/orientdb/utils/OrientDBTest.java [hunk 1]
```diff
@@ -6,11 +6,37 @@
...
  */
 public class OrientDBTest extends CRUDTest {
 
+	private static final String TEST_DB_URL = "memory:/tmp/orientdb_ogm_test";
+
+	/**
+	 * Drop and create the test classes (tables) schema in OrientDB
+	 */
+	@Before
+	public void intialize() {
```
Comment (DavideD) [inline orientdb/src/test/java/org/hibernate/datastore/ogm/orientdb/utils/OrientDBTest.java]: Is this required?

Comment (DavideD) [inline orientdb/src/test/java/org/hibernate/datastore/ogm/orientdb/utils/OrientDBTest.java]: Ah I see now, ideally we should do this when the session factory is created.
Using the `SchemaDefiner` I was talking about before

Comment (cristhiank) [inline orientdb/src/test/java/org/hibernate/datastore/ogm/orientdb/utils/OrientDBTest.java]: Thanks @DavideD  I will check the SchemaDefiner interface

**Comment by DavideD** [comment]
@cristhiank I haven't finished reviewing this. I'll continue tomorrow.
It looks good overall so far. Thanks

**Comment by cristhiank** [comment]
@DavideD  What do you think of this PR ?? Sergey emailed me to work together and I wanna know if can use this codebase to start ?

**Comment by DavideD** [comment]
I think this PR is OK, I tried to merge it but it seems that there some conflicts. Do you think we you can fix them?

**Pull request closed without merging**